Linux support + several changes to improve/simplify buildability in open-source

jamesjhedley opened the pull request:
-Linux support
-Adding node binaries to repo
-Adding logic to default to bbdnk installation tools if local dependencies/tools does not exist
-Improving/simplifying buildability in open-source

Thanks to @badtoyz for the assistance with the Linux implementation.

When testing, a simple smoke on both Windows and Mac should be sufficient, since we don't officially support Linux.

nukulb commented:
Bbwp should be changed to use system node if it doesn't find the binary

Just a thought
~Nukul

From: James Keshavarzi [mailto:notifications@github.com]
Sent: Wednesday, October 03, 2012 06:36 PM
To: blackberry-webworks/BB10-Webworks-Packager BB10-Webworks-Packager@noreply.github.com
Subject: [BB10-Webworks-Packager] Linux support + several changes to improve/simplify buildability in open-source (#89)

-Linux support
-Adding node binaries to repo
-Improving/simplifying buildability in open-source

---

You can merge this Pull Request by running:

  git pull https://github.com/blackberry-webworks/BB10-Webworks-Packager next-linux-support

Or view, comment on, or merge it at:

  https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89

Commit Summary
-   Adding node binaries to repo + improving/simplifying buildability in …
-   Adding linux Support.
-   Fixed issue where bbwp scripts in target/zip folder lost their execut…

File Changes
-   M .gitignore (3)
-   M Framework (2)
-   M bbwp (6)
-   M bbwp.bat (2)
-   M build/build/conf.js (1)
-   M build/build/pack.js (23)
-   M configure (4)
-   M lib/signing-helper.js (12)
-   M pom.xml (2)
-   A third_party/node/linux/readme (1)
-   A third_party/node/mac/node (0)
-   A third_party/node/windows/node.exe (0)

Patch Links
-   https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89.patch
-   https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89.diff

—
Reply to this email directly or view it on GitHubhttps://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89.

---

This transmission (including any attachments) may contain confidential information, privileged material (including material protected by the solicitor-client or other applicable privileges), or constitute non-public information. Any use of this information by anyone other than the intended recipient is prohibited. If you have received this transmission in error, please immediately reply to the sender and delete this information from your system. Use, dissemination, distribution, or reproduction of this transmission by unintended recipients is not authorized and may be unlawful.

badtoyz commented:
@nukulb i would not do that because the sdk uses and older node. i had to down grade my node on my system to try and get the sdk to build

nukulb commented:
Well I am hoping we can go to 0.8.11 very soon
~Nukul

From: Mike Robinson [mailto:notifications@github.com]
Sent: Wednesday, October 03, 2012 07:29 PM
To: blackberry-webworks/BB10-Webworks-Packager BB10-Webworks-Packager@noreply.github.com
Cc: Nukul Bhasin
Subject: Re: [BB10-Webworks-Packager] Linux support + several changes to improve/simplify buildability in open-source (#89)

@nukulbhttps://github.com/nukulb i would not do that because the sdk uses and older node. i had to down grade my node on my system to try and get the sdk to build

—
Reply to this email directly or view it on GitHubhttps://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9126066.

---

This transmission (including any attachments) may contain confidential information, privileged material (including material protected by the solicitor-client or other applicable privileges), or constitute non-public information. Any use of this information by anyone other than the intended recipient is prohibited. If you have received this transmission in error, please immediately reply to the sender and delete this information from your system. Use, dissemination, distribution, or reproduction of this transmission by unintended recipients is not authorized and may be unlawful.

badtoyz commented:
that would be better  0.6.10 is a bit old

nukulb commented:
By the 0.8.11 works just fine so I am not sure why you had an issue.

We are just waiting for wrench module to update for windows
~Nukul

From: Mike Robinson [mailto:notifications@github.com]
Sent: Wednesday, October 03, 2012 07:33 PM
To: blackberry-webworks/BB10-Webworks-Packager BB10-Webworks-Packager@noreply.github.com
Cc: Nukul Bhasin
Subject: Re: [BB10-Webworks-Packager] Linux support + several changes to improve/simplify buildability in open-source (#89)

that would be better 0.6.10 is a bit old

—
Reply to this email directly or view it on GitHubhttps://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9126140.

---

This transmission (including any attachments) may contain confidential information, privileged material (including material protected by the solicitor-client or other applicable privileges), or constitute non-public information. Any use of this information by anyone other than the intended recipient is prohibited. If you have received this transmission in error, please immediately reply to the sender and delete this information from your system. Use, dissemination, distribution, or reproduction of this transmission by unintended recipients is not authorized and may be unlawful.

badtoyz commented:
i will try it right now

badtoyz commented:
when i use 0.8.11 i get this error 
path.existsSync is now called `fs.existsSync`

but it still packages and signs ok

nukulb commented:
That is just a warning due to depreciation.

Bar should work fine.

Framework is on device so it is not truly run time dependent on node.

~Nukul

From: Mike Robinson [mailto:notifications@github.com]
Sent: Wednesday, October 03, 2012 08:43 PM
To: blackberry-webworks/BB10-Webworks-Packager BB10-Webworks-Packager@noreply.github.com
Cc: Nukul Bhasin
Subject: Re: [BB10-Webworks-Packager] Linux support + several changes to improve/simplify buildability in open-source (#89)

when i use 0.8.11 i get this error
path.existsSync is now called fs.existsSync

but it still packages and signs ok

—
Reply to this email directly or view it on GitHubhttps://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9127317.

---

This transmission (including any attachments) may contain confidential information, privileged material (including material protected by the solicitor-client or other applicable privileges), or constitute non-public information. Any use of this information by anyone other than the intended recipient is prohibited. If you have received this transmission in error, please immediately reply to the sender and delete this information from your system. Use, dissemination, distribution, or reproduction of this transmission by unintended recipients is not authorized and may be unlawful.

nukulb commented:
@jkeshavarzi - are we still restricted by any modules?

I was thinking that this change will require regression on the packager anyways and if we merge this change with moving to new node we can reduce testing effort.
plus we have community support for linux with @badtoyz for smoke tests anyways :)

nukulb commented:
@jkeshavarzi - actually lets just get this out first. I would like us to build in the open source as soon as possible and allow people to patch from feature branches.

jamesjhedley commented:
Ok, sounds good @nukulb.

jamesjhedley commented:
Just tested on Mac and Windows and I was able to configure/jake and run bbwp (sign and package app).
The new logic to default to bbndk tools if dependencies/tools is not available is working as well.
Going to get @tracyli to try now.

nukulb commented:
@jkeshavarzi can I get a squashed commit please? There are 6 commits here

jamesjhedley commented:
@nukulb squashed.

tracyli commented:
tested in mac and windows with James' help: configure & jake work great and smoke test widget can be packaged and run successfully

jamesjhedley commented:
@badtoyz Can you please give it a quick test on Linux and make sure it's still good to go?
@nukulb Are you finished with your code review?

badtoyz commented:
I will try it when I get home. I'm still new to git so how do I get this branch ?
-----Original Message-----
From: James Keshavarzi notifications@github.com
Date: Fri, 5 Oct 2012 09:09:26 
To: blackberry-webworks/BB10-Webworks-PackagerBB10-Webworks-Packager@noreply.github.com
Reply-To: blackberry-webworks/BB10-Webworks-Packager reply@reply.github.com
Cc: Mike Robinsonbadtoyz@gmail.com
Subject: Re: [BB10-Webworks-Packager] Linux support + several changes to improve/simplify buildability in open-source (#89)

@badtoyz Can you please give it a quick test on Linux and make sure it's still good to go?
@nukulb Are you finished with your code review?

---

Reply to this email directly or view it on GitHub:
https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9180719

jamesjhedley commented:
@badtoyz 

`git clone https://github.com/blackberry-webworks/BB10-Webworks-Packager.git
cd BB10-Webworks-Packager
git checkout next-linux-support`

badtoyz commented:
is there a dependencies folder we are working from or can i just use the one from the sdk and just change where the node.linux / node.mac to /mac/node /linux/node ?

badtoyz commented:
ok i see it all got move to the third_party folder

jamesjhedley commented:
@badtoyz Yup yup. Everything should be good to go, if you checkout next-linux-support. Also, you can grab the dependencies folder from a sdk build like you mentioned :).

jamesjhedley commented:
Also note, you will need to manually copy the linux binary to third_party/node/linux. We won't be storing the linux binaries in the repo, since they can potentially differ per distro.

badtoyz commented:
when i compiled node i made it compile to a generic system so it should across 32/64bit systems

badtoyz commented:
i am getting this error and it just keeps looping

Cloning into dependencies/webplatform...
error: Couldn't resolve host 'github.rim.net' while accessing http://github.rim.net/webworks/webplatform.git/info/refs

fatal: HTTP request failed
Clone of 'http://github.rim.net/webworks/webplatform.git' into submodule path 'dependencies/webplatform' failed
./configure: line 10: cd: ./dependencies/webplatform/framework: No such file or directory

jamesjhedley commented:
You don't have access to that submodule. I am going to be making changes next week to make this submodule optional. For now, can you please comment-out the below lines in the `Framework/configure` script.

cd ./dependencies/webplatform/framework;
if [[ "$OS" == _Windows_ ]]; then
    source configure;
else
    ./configure
fi
jake

badtoyz commented:
ok took that out and i still get this error

Cloning into dependencies/webplatform...
error: Couldn't resolve host 'github.rim.net' while accessing http://github.rim.net/webworks/webplatform.git/info/refs

fatal: HTTP request failed
Clone of 'http://github.rim.net/webworks/webplatform.git' into submodule path 'dependencies/webplatform' failed

is it because it is still asking for the git submodule update --init;

jamesjhedley commented:
Yea lets get rid of that too. Should get us farther. I think we might run into some problems when you build though [run jake], since you don't have the webplatform. But we shall tackle that if it becomes an issue.

badtoyz commented:
ok i for got to setup the build system on this system. i will have to do that now

jamesjhedley commented:
Ok, i managed to get everything building without the web platform submodule, using the below steps.

I removed the code that i mentioned before from Framework/configure

After you do this, run ./configure on the packager and you will get an error like

`Error: ENOENT, no such file or directory '/Users/jkeshavarzi/clones/bwpNoWebPlat/Framework/dependencies/webplatform/framework/clientFiles/ui-resources'
    at Object.statSync (fs.js:414:18)
(See full trace by running task with --trace)`

What you need to do at this point is navigate to Framework/dependencies/webplatform and create a folder under this called "framework" and under framework create another folder called "clientFiles". In the clienFiles directory, you need to copy a few files that you won't have because you don't have access to the web platform submodule. Please download our latest released SDK [1.0.2.9] and copy the following files into the newly created clientFiles directory.

Framework/dependencies/bootstrap/webplatform.js
Framework/ui-resources [folder]
Also create a file called i18n.js. Just leave it empty. Shouldn't cause any issues i don't think.

Once these files are in place, re-run ./configure on the BB10-Webworks-Packager.

Should get a BUILD SUCCESS

Then run jake and navigate to the target/zip folder.
This is the output of the build. Use it like you would the regular SDK.
I.e. bbwp ~/app.zip -o ~/bin

Let me know if you have any issues.

Thanks again for helping us test this!

badtoyz commented:
ok i will try it now

badtoyz commented:
im getting a lot of build errors and im thinking i have an old ndk on my system i will update my ndk now

jamesjhedley commented:
Ok, hopefully the latest "released" bbndk works for you.

badtoyz commented:
im getting a lot of errors in the Framework/ext/push/

/home/mike/bbndk-10.0.4-beta/target/qnx6/usr/include/bb/communications/push/PushListener.hpp:33:18: note:   virtual void bb::communications::push::PushListener::pushNotification(const bb::communications::push::PushPayload&)
/home/mike/test/BB10-Webworks-Packager/Framework/ext/push/native/push_ndk.cpp: At global scope:
/home/mike/test/BB10-Webworks-Packager/Framework/ext/push/native/push_ndk.cpp:229:36: error: variable or field 'onPushTransportReady' declared void
/home/mike/test/BB10-Webworks-Packager/Framework/ext/push/native/push_ndk.cpp:229:36: error: 'PushCommand' was not declared in this scope
/home/mike/test/BB10-Webworks-Packager/Framework/ext/push/native/push_ndk.cpp: In member function 'void webworks::PushNDK::MonitorMessages()':
/home/mike/test/BB10-Webworks-Packager/Framework/ext/push/native/push_ndk.cpp:262:33: error: 'webworks::PushService' has no member named 'processMsg'
cc: /home/mike/bbndk-10.0.4-beta/host/linux/x86/usr/lib/gcc/i486-pc-nto-qnx8.0.0/4.6.3/cc1plus error 1
make[2]: **\* [ext/push/native/CMakeFiles/pushJNext.dir/push_js.cpp.o] Error 1
make[2]: **\* Waiting for unfinished jobs....
cc: /home/mike/bbndk-10.0.4-beta/host/linux/x86/usr/lib/gcc/i486-pc-nto-qnx8.0.0/4.6.3/cc1plus error 1
make[2]: **\* [ext/push/native/CMakeFiles/pushJNext.dir/push_ndk.cpp.o] Error 1
make[1]: **\* [ext/push/native/CMakeFiles/pushJNext.dir/all] Error 2
make: **\* [all] Error 2
BB10-Framework BUILD FAILED:
2

badtoyz commented:
nevermind i did not update my system variables. it is till linked to the old bbndk

badtoyz commented:
ok now that i corrected all my messups it done it compiled for linux

[100%]
Built target dialog
BB10-Framework BUILD SUCCESS

badtoyz commented:
ok just packaged and signed a bar file all seams well

nukulb commented:
Awesome ness

James let's document these steps ad make sure thy get into our readme pages
~Nukul

From: Mike Robinson [mailto:notifications@github.com]
Sent: Friday, October 05, 2012 06:40 PM
To: blackberry-webworks/BB10-Webworks-Packager BB10-Webworks-Packager@noreply.github.com
Cc: Nukul Bhasin
Subject: Re: [BB10-Webworks-Packager] Linux support + several changes to improve/simplify buildability in open-source (#89)

ok just packaged and signed a bar file all seams well

—
Reply to this email directly or view it on GitHubhttps://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9192043.

---

This transmission (including any attachments) may contain confidential information, privileged material (including material protected by the solicitor-client or other applicable privileges), or constitute non-public information. Any use of this information by anyone other than the intended recipient is prohibited. If you have received this transmission in error, please immediately reply to the sender and delete this information from your system. Use, dissemination, distribution, or reproduction of this transmission by unintended recipients is not authorized and may be unlawful.

nukulb commented:
r+ for this

but I think this needs to merged into blackberry:master and then circle merged blackberry-webworks:next

jamesjhedley commented:
@nukulb Made the changes as requested. We now copy bbndk tools to the build folder instead of "using" the tools from their original location.

Can you please review.

nukulb commented:
got this 

```
BB10-Webworks BUILD FAILED:
ENOENT, no such file or directory '/Users/nukulb/github-reviews/packager/dependencies/tools'
Error: ENOENT, no such file or directory '/Users/nukulb/github-reviews/packager/dependencies/tools'
    at Object.statSync (fs.js:424:18)
    at Object.copyDirSyncRecursive (/Users/nukulb/github-reviews/packager/node_modules/wrench/lib/wrench.js:175:23)
    at copyFolder (/Users/nukulb/github-reviews/packager/build/build/pack.js:28:12)
    at /Users/nukulb/github-reviews/packager/build/build/pack.js:55:5
    at Object.pass (/Users/nukulb/github-reviews/packager/node_modules/jWorkflow/lib/jWorkflow.js:41:52)
    at Object.pass (/Users/nukulb/github-reviews/packager/node_modules/jWorkflow/lib/jWorkflow.js:44:44)
    at Object.start (/Users/nukulb/github-reviews/packager/node_modules/jWorkflow/lib/jWorkflow.js:119:32)
    at Function.<anonymous> (/Users/nukulb/github-reviews/packager/build/build.js:46:11)
    at [object Object].action (/Users/nukulb/github-reviews/packager/build/build.js:35:18)
    at [object Object].run (/opt/local/lib/node_modules/jake/lib/task/task.js:207:21)

```

jamesjhedley commented:
You sure you have the latest commit? I just tested this on mac and it was fine.

badtoyz commented:
I did a git fetch on the packager and the framework and built fine on linux
-----Original Message-----
From: James Keshavarzi notifications@github.com
Date: Wed, 10 Oct 2012 10:30:31 
To: blackberry-webworks/BB10-Webworks-PackagerBB10-Webworks-Packager@noreply.github.com
Reply-To: blackberry-webworks/BB10-Webworks-Packager reply@reply.github.com
Cc: Mike Robinsonbadtoyz@gmail.com
Subject: Re: [BB10-Webworks-Packager] Linux support + several changes to
 improve/simplify buildability in open-source (#89)

You sure you have the latest commit? I just tested this on mac and it was fine.

---

Reply to this email directly or view it on GitHub:
https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9312145

jamesjhedley commented:
@badtoyz Did you also `git pull` to get the latest changes? Do a `git log` and make sure you have my latest commit before testing "copy bbndk tools to build folder if they don't exist in repo". Thanks :).

badtoyz commented:
Would it be best if I started from scratch again to see if every thing is in place. Because my changes are stiil there from the last test

-----Original Message-----
From: James Keshavarzi notifications@github.com
Date: Wed, 10 Oct 2012 12:18:25 
To: blackberry-webworks/BB10-Webworks-PackagerBB10-Webworks-Packager@noreply.github.com
Reply-To: blackberry-webworks/BB10-Webworks-Packager reply@reply.github.com
Cc: Mike Robinsonbadtoyz@gmail.com
Subject: Re: [BB10-Webworks-Packager] Linux support + several changes to
 improve/simplify buildability in open-source (#89)

@badtoyz Did you also `git pull` to get the latest changes? Do a `git log` and make sure you have my latest commit before testing "copy bbndk tools to build folder if they don't exist in repo". Thanks :).

---

Reply to this email directly or view it on GitHub:
https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9315831

jamesjhedley commented:
Which ever approach is easiest for you. Should be good if you just `git pull`, but if you want to be extra safe you can begin from scratch. We didn't change too much. You would just need to remove the lines from the configure script again and copy over the dependencies folder you created earlier (with files from the SDK installation).

badtoyz commented:
Ok did a git pull on the packager and when I do a git pull on the framework it asks for a branch should I be on master or next ?

-----Original Message-----
From: James Keshavarzi notifications@github.com
Date: Wed, 10 Oct 2012 12:43:11 
To: blackberry-webworks/BB10-Webworks-PackagerBB10-Webworks-Packager@noreply.github.com
Reply-To: blackberry-webworks/BB10-Webworks-Packager reply@reply.github.com
Cc: Mike Robinsonbadtoyz@gmail.com
Subject: Re: [BB10-Webworks-Packager] Linux support + several changes to
 improve/simplify buildability in open-source (#89)

Which ever approach is easiest for you. Should be good if you just `git pull`, but if you want to be extra safe you can begin from scratch. We didn't change too much. You would just need to remove the lines from the configure script again and copy over the dependencies folder you created earlier (with files from the SDK installation).

---

Reply to this email directly or view it on GitHub:
https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9316703

badtoyz commented:
Did the git pull and every thing still seams good on linux
-----Original Message-----
From: James Keshavarzi notifications@github.com
Date: Wed, 10 Oct 2012 12:43:11 
To: blackberry-webworks/BB10-Webworks-PackagerBB10-Webworks-Packager@noreply.github.com
Reply-To: blackberry-webworks/BB10-Webworks-Packager reply@reply.github.com
Cc: Mike Robinsonbadtoyz@gmail.com
Subject: Re: [BB10-Webworks-Packager] Linux support + several changes to
 improve/simplify buildability in open-source (#89)

Which ever approach is easiest for you. Should be good if you just `git pull`, but if you want to be extra safe you can begin from scratch. We didn't change too much. You would just need to remove the lines from the configure script again and copy over the dependencies folder you created earlier (with files from the SDK installation).

---

Reply to this email directly or view it on GitHub:
https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9316703

jamesjhedley commented:
Yea you want to be on next and doing a `git pull origin next` on the BB10-Webworks-Packager repo should be enough.

badtoyz commented:
Just did the git pull on the framework to next and it builds just fine

Sent from my BlackBerry® PlayBook™
www.blackberry.com

---

_From:_ "James Keshavarzi" notifications@github.com
_To:_ "blackberry-webworks/BB10-Webworks-Packager" <
BB10-Webworks-Packager@noreply.github.com>
_CC:_ "Mike Robinson" badtoyz@gmail.com
_Sent:_ October 10, 2012 4:27 PM
_Subject:_ Re: [BB10-Webworks-Packager] Linux support + several changes to
improve/simplify buildability in open-source (#89)

Yea you want to be on next and doing a git pull origin next on the
BB10-Webworks-Packager repo should be enough.

—
Reply to this email directly or view it on
GitHubhttps://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9318244.

jamesjhedley commented:
You mean the Packager? Or both? Both would be good too.

badtoyz commented:
I just did it on both
-----Original Message-----
From: James Keshavarzi notifications@github.com
Date: Wed, 10 Oct 2012 13:39:53 
To: blackberry-webworks/BB10-Webworks-PackagerBB10-Webworks-Packager@noreply.github.com
Reply-To: blackberry-webworks/BB10-Webworks-Packager reply@reply.github.com
Cc: Mike Robinsonbadtoyz@gmail.com
Subject: Re: [BB10-Webworks-Packager] Linux support + several changes to
 improve/simplify buildability in open-source (#89)

You mean the Packager? Or both? Both would be good too.

---

Reply to this email directly or view it on GitHub:
https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9318670

jamesjhedley commented:
Awesome. Thanks for all the help @badtoyz . Very much appreciated!

badtoyz commented:
Not a problem at all. If you need any more help just ask and I will do what I can
-----Original Message-----
From: James Keshavarzi notifications@github.com
Date: Wed, 10 Oct 2012 13:44:33 
To: blackberry-webworks/BB10-Webworks-PackagerBB10-Webworks-Packager@noreply.github.com
Reply-To: blackberry-webworks/BB10-Webworks-Packager reply@reply.github.com
Cc: Mike Robinsonbadtoyz@gmail.com
Subject: Re: [BB10-Webworks-Packager] Linux support + several changes to
 improve/simplify buildability in open-source (#89)

Awesome. Thanks for all the help @badtoyz . Very much appreciated!

---

Reply to this email directly or view it on GitHub:
https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/89#issuecomment-9318832

nukulb commented:
r+ it seemed to work me. I didn't have the latest commit.

Commit:
Linux support + several changes to improve/simplify buildability in open-source

List of changes:
-Adding node binaries to repo
-Adding linux Support.
-Fixed issue where bbwp scripts in target/zip folder lost their execute permissions + ensure we 0755 node binaries and bbdnk tools after jake.
-Add logic to include dependencies/tools in target/zip if it exists,
when running jake [Open-source builds]
-Adding logic to copy/use bbdnk installation tools if local dependencies/tools does not exist

Reviewed By: Nukul Bhasin <nbhasin@rim.com>
Tested By: Tracy Li <tli@rim.com>

jamesjhedley commented:
Rebased + squashed.

@tracyli Can you retest on Mac and Windows please :).

tracyli commented:
tested in mac and windows: configure & jake work great and smoke test widget can be packaged and run without problem

nukulb commented:
circle merged into blackberry:next , then blackberry:master then blackberry-webworks:master and then blackberry-webworks:next
